cpp/src/fil/common.cuh

```diff
@@ -38,81 +38,92 @@ __host__ __device__ __forceinline__ int forest_num_nodes(int num_trees, int dept
   return num_trees * tree_num_nodes(depth);
 }
 
+template <typename F>
```

Comment:
Thinking of this further, a name like `float_t` or `real_t` for this type parameter would be better. This applies throughout this pull request.

Reply:
Done.